lib/ruby_indexer/lib/ruby_indexer/rbs_indexer.rb

```diff
@@ -47,6 +49,32 @@ def handle_class_declaration(declaration, pathname)
       comments = Array(declaration.comment&.string)
       parent_class = declaration.super_class&.name&.name&.to_s
       class_entry = Entry::Class.new(nesting, file_path, location, comments, parent_class)
+
+      # We also need to record the class's mixins
+      declaration.each_mixin do |mixin|
+        name = mixin.name.name.to_s
+        mixin_operation =
+          case mixin
+          when RBS::AST::Members::Include
+            Entry::Include.new(name)
+          when RBS::AST::Members::Extend
+            Entry::Extend.new(name)
+          when RBS::AST::Members::Prepend
+            Entry::Prepend.new(name)
+          end
+        class_entry.mixin_operations << mixin_operation if mixin_operation
+      end
+
+      @index << class_entry
+    end
+
+    sig { params(declaration: RBS::AST::Declarations::Module, pathname: Pathname).void }
+    def handle_module_declaration(declaration, pathname)
```

Comment:
We need the exact same code that was added to `handle_class_declaration` here, since modules themselves can `include`, `prepend` or `extend` other modules.

Reply:
I added `add_declaration_mixins_to_entry` for it. But atm we can't write test for it yet as no core modules have mixin themselves.